### PR TITLE
Add optimized_for_gmon to optimize circuits for Sycamore devices

### PR DESCRIPTION
**Pull request by dstrain115**
- Add optimize_for_gmon that can take a string ('xmon',
'xmon_partial_cz', 'sqrt_iswap', 'sycamore') and run the appropriate
optimizers on it.
- Convert optimize_for_xmon to use this function and mark it as
deprecated.

### CODE REVIEW

File: cirq/google/optimizers/optimize_for_gmon.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+
+def optimized_for_gmon(circuit: 'cirq.Circuit',
+                       new_device: Optional['cirq.google.XmonDevice'] = None,
+                       qubit_map: Callable[['cirq.Qid'], devices.GridQubit] =
+                       lambda e: cast(devices.GridQubit, e),
+                       optimizer_type: str = 'sqrt_iswap') -> 'cirq.Circuit':
+    """Optimizes a circuit with XmonDevice in mind.
+
+  Starts by converting the circuit's operations to the xmon gate set, then
```
Comment (kevinsung) [inline cirq/google/optimizers/optimize_for_gmon.py]: It seems like xmon and gmon are being conflated here.

Comment (dstrain115) [reply cirq/google/optimizers/optimize_for_gmon.py]: Updated pydocs.

File: cirq/google/optimizers/optimize_for_gmon.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    'sycamore': _SYCAMORE_OPTIMIZERS,
+}
+
+
+def optimized_for_gmon(circuit: 'cirq.Circuit',
+                       new_device: Optional['cirq.google.XmonDevice'] = None,
+                       qubit_map: Callable[['cirq.Qid'], devices.GridQubit] =
+                       lambda e: cast(devices.GridQubit, e),
+                       optimizer_type: str = 'sqrt_iswap') -> 'cirq.Circuit':
+    """Optimizes a circuit for Google devices.
```
Comment (kevinsung) [inline cirq/google/optimizers/optimize_for_gmon.py]: In this case the function name shouldn't refer specifically to gmon.

Comment (dstrain115) [reply cirq/google/optimizers/optimize_for_gmon.py]: +Craig.  Any preference on name?  optimize_for_google()?

Comment (Strilanc) [reply cirq/google/optimizers/optimize_for_gmon.py]: `optimize_for_google` will go out of date quickly. Maybe `optimize_for_sycamore`.

File: cirq/google/optimizers/optimize_for_xmon.py [hunk 3]
```diff
@@ -14,72 +14,26 @@
...
-                                 allow_partial_czs=True).optimize_circuit,
-    _merge_rots,
-    optimizers.EjectPhasedPaulis(tolerance=_TOLERANCE).optimize_circuit,
-    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
-    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
-]
-
 
+@deprecated(deadline='v0.7.0',
+            fix='Use optimized_for_gmon with '
```
Comment (Strilanc) [inline cirq/google/optimizers/optimize_for_xmon.py]: This method does something different than optimized_for_gmon, so it's not really appropriate to just substitute one in.

Comment (dstrain115) [reply cirq/google/optimizers/optimize_for_xmon.py]: Removed deprecation warning, but it still depends on the underlying function.

Also changed the name to optimize_for_sycamore.

**Pull request merged**